app/controllers/api/v1/owners_controller.rb

```diff
@@ -1,7 +1,7 @@
 class Api::V1::OwnersController < Api::BaseController
-  before_action :authenticate_with_api_key, except: %i[show gems]
+  before_action :find_api_key, only: %i[create destroy]
```

Comment:
Prefer secure default here, so using `except` instead of `only`. This would make the default for any actions added in the future to be secure:

```diff
# Newly added actions will not require API key authentication,
# could lead to introducing vulnerability accidentally. Make
# it so actions have to opt-out of authentication:
-before_action :find_api_key, only: %i[create destroy]
+before_action :find_api_key, except: %i[show gems]
```

Also consider renaming from `find_api_key` as it does more than find, it also performs authentication, its main responsibility 

Just now reviewing this controller I thought it didn't have authentication due to this name, and I expect I'll keep making that mistake on subsequent reviews if it isn't explicit about performing authentication.